hw_3/Finder.js

```diff
@@ -0,0 +1,227 @@
+
+                    const indexSubstr = tmp.indexOf(this._search);
+                    if(indexSubstr != -1) {
+
+                        const beforeSubstr = tmp.substr(0, indexSubstr-1);
+                        const afterSubstr = tmp.substr(indexSubstr+this._search.length, 20);
+
+                        rs.destroy();
+
+                        resolve({
+                            beforeSubstr,
+                            afterSubstr,
+                        });
+                    } else {
+                        tmp = tmp.slice(prevChunkLength);
+                        prevChunkLength = chunk.length;
+                    }
+                }
+            });
+
+            rs.on("close", () => {
+                resolve();
+            });
+        });
+    }
+
+    _readDir(path, options) {
+        return new Promise((resolve, reject) => {
+            readdir(path, options, (err, res) => {
+                if (err) {
+                    return reject(`Error reading the directory: ${path}`);
+                }
+
+                resolve(res);
+            });
+        });
+    }
+
+    _setTimer() {
+        if (this._timer) {
+            clearTimeout(this._timer);
+        }
+
+        this._timer = setTimeout(() => {
+            this.emit('processing', {
+                dir: this._dirProcessCount,
+                files: this._fileProcessCount,
+            });
+        }, this._interval);
+    }
+
+    _writeLogFile(beforeSubstr, afterSubstr) {
+        return new Promise((resolve, reject) => {
+            const ws = createWriteStream(`./log/${this._logFileName}.txt`, {
+                flags: "a"
+            });
+
+            ws.on("finish", ()=> {
+                resolve();
+            });
+
+            ws.write(`${beforeSubstr}${afterSubstr}`);
+
+            ws.end();
+        });
+    }
+
+    _isFileTypeMatched(ft, file) {
+        const extWithoutDot = this._ext.replace(".", "");
+        const currExt = extname(file.name);
+        return (ft && ft.ext === extWithoutDot)
+            ||
+            (!ft && this._ext === currExt);
+    }
+
+    _isFileText(ft, ext) {
+        const mime = ["plain/text", "text/rtf"];
+        const extList = [".json", ".js", ".css", ".txt"];
+
+        if(ft) {
+            return mime.includes(ft.mime);
+        } else {
+            return extList.includes(ext);
+        }
+    }
+
+    _checkFileType(element, elementPath) {        
+        return new Promise((resolve, reject) => {
+            const rs = createReadStream(elementPath, {
+                highWaterMark: 4100,
+            });
+
+            rs.on("data", (chunk) => {
+                FileType.fromBuffer(chunk).then((ft) => {
+                    resolve({
+                        isFileTypeMatched: this._isFileTypeMatched(ft, element),
+                        ft,
+                    });
+                }).catch((e) => {
+                    reject(e);
+                });
+
+                rs.close();
```

Comment:
прям в документации они пишут, что это плохой метод https://nodejs.org/docs/latest-v12.x/api/fs.html#fs_fs_createreadstream_path_options
```suggestion
                rs.destroy();
```